tests/mir_test_doubles/file_logger.cpp

```diff
@@ -25,23 +25,23 @@
 namespace ml = mir::logging;
 namespace mtd = mir::test::doubles;
 
-mtd::StreamLogger::StreamLogger(std::unique_ptr<std::ostream> stream)
-: out(std::move(stream))
+mtd::FileLogger::FileLogger(std::ofstream stream)
+: file{std::move(stream)}
 {
-    if (!out || !out->good())
+    if (!file.good())
     {
-        BOOST_THROW_EXCEPTION(std::runtime_error{"Received invalid stream for logging"});
+        BOOST_THROW_EXCEPTION(std::runtime_error{"Received invalid file stream for logging"});
     }
 }
 
-void mtd::StreamLogger::log(mir::logging::Severity severity,
+void mtd::FileLogger::log(mir::logging::Severity severity,
                           const std::string& message,
                           const std::string& component)
 {
-    if (!out->good())
+    if (!file.good())
     {
-        BOOST_THROW_EXCEPTION(std::runtime_error{"Failed to write to log stream"});
+        BOOST_THROW_EXCEPTION(std::runtime_error{"Failed to write to log file"});
```

Comment:
Another thought: It really a good idea to throw exceptions when logging fails?

Maybe this is arguably OK for test code, but not a good precedent.